docs/dev/dev_setup.rst

```diff
@@ -0,0 +1,217 @@
+Create the PUDL Development Environment
+-------------------------------------------------------------------------------
+Inside the ``devtools`` directory of your newly cloned repository, you'll see an
+``environment.yml`` file, which specifies the ``pudl-dev`` ``conda`` environment.  You
+can create and activate that environment from within the main repository directory by
+running:
+
+.. code-block:: console
+
+    $ conda update conda
+    $ conda env create --name pudl-dev --file devtools/environment.yml
+    $ conda activate pudl-dev
+
+This environment installs the ``catalystcoop.pudl`` package directly using the code in
+your cloned repository so that it can be edited during development. It also installs
+all of the software PUDL depends on, some packages for testing and quality control,
+working with interactive Jupyter Notebooks, and a few Python packages that have binary
+dependencies which can be easier to satisfy through ``conda`` packages.
+
+-------------------------------------------------------------------------------
+Update the PUDL Development Environment
+-------------------------------------------------------------------------------
+Periodically you will need to update your development software environment. This will
+get you newer versions of existing dependencies, and also incorporate any changes that
+have been made to the environment specification by others. The most reliable way to do
+this is to remove the existing environment and recreate it. From within the top
+directory of the PUDL repository:
+
+.. code-block:: console
+
+    $ git checkout BRANCH_NAME
+    $ git pull
+    $ conda deactivate
+    $ conda update conda
+    $ conda env remove --name pudl-dev
+    $ conda env create --name pudl-dev --file devtools/environment.yml
+    $ conda activate pudl-dev
+
+.. Note::
+
+    If you are switching back and forth between different branches within the
+    repository, and they have significantly different software environments specified,
+    you may need to recreate the environment after switching from one to the other. In
+    this case it may be more convenient to create two separate environments with
+    different names that you can switch between. This arrangement should be rare.
+
+If you find yourself recreating the environment frequently, and are frustrated by how
+long it takes ``conda`` to solve the dependencies, we recommend using the
+`mamba <https://github.com/mamba-org/mamba>`__ solver. In your ``base`` conda
+environment (i.e. with no conda environment activated) you can install it with:
+
+.. code-block:: console
+
+    $ conda install mamba
+
+Then the above development environment update process would become:
+
+.. code-block:: console
+
+    $ git checkout BRANCH_NAME
+    $ git pull
+    $ conda deactivate
+    $ mamba update mamba
+    $ mamba env remove --name pudl-dev
+    $ mamba env create --name pudl-dev --file devtools/environment.yml
+    $ conda activate pudl-dev
```

Comment:
just checking that conda and mamba can co-exist like this

Reply:
Yes, one still has to use `conda` for environment activation & deactivation, but `mamba` can do all the software installation / updating / dependency solving.  They're both installed at once, and either one of them can do those jobs. Just for some reason the environment activation has to be done with `conda` -- and you'll get a message to that effect if you try and use `mamba` to do it instead.